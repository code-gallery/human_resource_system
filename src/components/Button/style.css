.Button {
  all: initial;
  box-sizing: border-box;
  display: block;
  width: 100%;
  padding: 17px;
  font-family: 'Graphik-Medium', sans-serif;
  font-size: 1rem;
  font-weight: 500;
  -webkit-font-smoothing: antialiased;
  line-height: 1;
  text-align: center;
  border-radius: 50px;
  cursor: pointer;
  color: #fff;
  transition: background-color 225ms ease-in;
}

.Button.is-disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

/** Green */
.Button--green {
  background-color: #72d371;
}

.Button--green:hover {
  background-color: #4CAD4B;
}

.Button--green.Button.is-disabled:hover {
  background-color: #72d371;
}

/** Blue */
.Button--blue {
  background-color: #498df0;
}

.Button--blue:hover {
  background-color: #2367CA;
}

.Button--blue.is-disabled:hover {
  background-color: #498df0;
}

/** Red */
.Button--red {
  background-color: #FB7C7C;
}

.Button--red:hover {
  background-color: #E26363;
}

.Button--red.is-disabled:hover {
  background-color: #FB7C7C;
}

/** Yellow */
.Button--yellow {
  background-color:#ffc40f;
}

.Button--yellow:hover {
  background-color: #edc13b;
}

.Button--yellow.is-disabled:hover {
  background-color:#ffc40f;
}

/** Amber */
.Button--amber {
  background-color:#ffc000;
}

.Button--amber:hover {
  background-color: #edc13b;
}

.Button--amber.is-disabled:hover {
  background-color:#ffc000;
}

.Button__icon {
  position: relative;
  top: 3px;
  display: inline-block;
  width: 1.0625rem;
  margin-right: 10px;
}
