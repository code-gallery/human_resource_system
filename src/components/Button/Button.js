import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import Icon from 'components/Icon'
import './style.css'

const Button = ({
  color,
  icon,
  className,
  children,
  onClick,
  disabled,
  type
}) => {
  const style = classNames(
    'Button',
    `Button--${color}`,
    { 'is-disabled': disabled },
    className
  )

  const handler = !disabled ? onClick : null

  return (
    <button className={style} onClick={handler} type={type} disabled={disabled}>
      {
        icon && (
          <span className="Button__icon">
            <Icon type={icon} color="#fff" />
          </span>
        )
      }
      {children}
    </button>
  )
}

Button.defaultProps = {
  color: 'green',
  icon: false,
  className: '',
  disabled: false,
  type: 'button'
}

Button.propTypes = {
  color: PropTypes.oneOf([ 'green', 'blue', 'red', 'amber', 'white' ]),
  icon: PropTypes.oneOfType([ PropTypes.string, PropTypes.bool ]),
  onClick: PropTypes.func.isRequired,
  className: PropTypes.string,
  children: PropTypes.node.isRequired,
  disabled: PropTypes.bool,
  type: PropTypes.string
}

export default Button
