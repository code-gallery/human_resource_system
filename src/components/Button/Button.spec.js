import React from 'react'
import renderer from 'react-test-renderer'
import Button from './index'

describe('<Button />', () => {
  const props = {
    onClick: jest.fn()
  }

  it('renders correctly', () => {
    const tree = renderer.create(
      <Button {...props}>Click Me</Button>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
