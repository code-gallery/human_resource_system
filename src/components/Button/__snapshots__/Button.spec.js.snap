// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<button
  className="Button Button--green"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Click Me
</button>
`;
